tuplex/codegen/include/ASTAnnotation.h

```diff
@@ -382,6 +388,9 @@ class ASTAnnotation {
     ///! traced types
     std::vector<python::Type> types;
 
+    ///! iterator-specific info
+    IteratorInfo *iteratorInfo;
```

Comment:
maybe use std::shared_ptr etc. for memory mangement? Avoid potentially also issues with weak vs. deep copy.

Reply:
i.e., is the memory of IteratorInfo etc. released properly?

Reply:
Refactored to std::shared_ptr.